File: MdExplorer/client2/src/app/md-explorer/models/IFileInfoNode.ts
// Segment of a compacted folder path (for VS Code-style compact folders)
export interface CompactSegment {
  name: string;       // Single folder name: "src", "main", "java"
  fullPath: string;   // Full path: "C:\project\src\main\java"
  level: number;      // Level in original tree
}

// implementation of this interface is MDFile
export interface IFileInfoNode {
  name: string;
  path: string;
  relativePath: string;
  fullPath: string;
  type: string;
  level: number;
  expandable: boolean;
  isLoading: boolean;
  childrens?: IFileInfoNode[];
  index: number;
  isIndexed?: boolean;
  indexingStatus?: 'idle' | 'indexing' | 'completed';
  indexingProgress?: number; // 0-100
  developmentTags?: string[]; // Tags for development folders (e.g., "program", "tests", "docs")

  // True when the folder owns a generated TOC file (<dirname>.md.directory)
  hasToc?: boolean;

  // True when the folder contains content the md-tree does not show (non-.md files or
  // markdown-empty subfolders). Drives the "reveal content" (eye) entry in the context menu.
  hasExtraContent?: boolean;
  // True once the extra content of this folder has been pulled into the tree (eye toggled on).
  extraLoaded?: boolean;
  // True on nodes that were injected by a folder "reveal" call (so a "hide" can remove them
  // again without touching the folder's real .md children).
  isExtra?: boolean;

  // Compact folder properties (VS Code-style)
  isCompacted?: boolean;                // True if this node is the result of compacting multiple folders
  compactedPath?: string;               // Compacted display path: "src / main / java"
  compactedSegments?: CompactSegment[]; // Segments for context menu selection

  // External app embedding
  appId?: string;
  appExecutable?: string;
  appArgs?: string[];
  appIcon?: string;
  appDescription?: string;
}

File: MdExplorer.Abstractions/Interfaces/IFileInfoNode.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MdExplorer.Abstractions.Interfaces
{
    public interface IFileInfoNode
    {
        public string Path { get; set; }
        public string FullPath { get; set; }
        public string RelativePath { get; set; }
        public string Name { get; set; }
        public string Type { get; set; }
        public int Level { get; set; }
        public bool Expandable { get; set; }
        IList<IFileInfoNode> Childrens { get; set; }

        // Nuove proprietà per caricamento incrementale
        bool IsIndexed { get; set; }
        string IndexingStatus { get; set; }
        int? IndexingProgress { get; set; }

        // Development tags per classificare le cartelle
        List<string> DevelopmentTags { get; set; }

        // True when the folder owns a generated TOC file (<dirname>.md.directory)
        bool HasToc { get; set; }

        // True when the folder contains content the md-tree does not show:
        // a direct non-.md file, or a direct subfolder dropped for lacking .md descendants.
        // Drives the "eye" reveal toggle on the folder node.
        bool HasExtraContent { get; set; }

        // External app embedding
        string AppId { get; set; }
        string AppExecutable { get; set; }
        List<string> AppArgs { get; set; }
        string AppIcon { get; set; }
        string AppDescription { get; set; }
    }
}
